android_app/app/src/main/java/com/health/openscale/gui/MainActivity.java

```diff
@@ -651,10 +657,92 @@ public void handleMessage(Message msg) {
                         Timber.e("Bluetooth scale message error: " + ex);
                     }
                     break;
+                case CHOOSE_SCALE_USER:
+                    chooseScaleUser(msg);
+                    break;
+                case ENTER_SCALE_USER_CONSENT:
+                    enterScaleUserConsent(msg);
+                    break;
             }
         }
     };
 
+    private void chooseScaleUser(Message msg) {
+        AlertDialog.Builder mBuilder = new AlertDialog.Builder(MainActivity.this);
+        Pair<CharSequence[], int[]> choices = (Pair<CharSequence[], int[]>)msg.obj;
+
+        mBuilder.setTitle(getResources().getString(R.string.info_select_scale_user));
+        mBuilder.setSingleChoiceItems(choices.first, -1 , new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialoginterface, int i) {
+                    Timber.d("UI selected " + i + ": " + choices.first[i] + " P-0" + choices.second[i]);
+                    OpenScale.getInstance().setBluetoothDeviceUserIndex(OpenScale.getInstance().getSelectedScaleUser().getId(), choices.second[i], callbackBtHandler);
+                    dialoginterface.dismiss();
+                }
+            });
+        mBuilder.setNegativeButton(getResources().getString(R.string.label_cancel), new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialoginterface, int i) {
+                    dialoginterface.dismiss();
+                }
+            });
+
+        AlertDialog mDialog = mBuilder.create();
+        mDialog.show();
+    }
+
+    private void enterScaleUserConsent(Message msg) {
+        final int appUserId = msg.arg1;
+        final int scaleUserIndex = msg.arg2;
+        final int[] consentCode = {-1};
+
+        AlertDialog.Builder mBuilder = new AlertDialog.Builder(MainActivity.this);
+        mBuilder.setTitle(getResources().getString(R.string.info_enter_consent_code_for_scale_user, Integer.toString(scaleUserIndex)));
+
+        final EditText input = new EditText(this);
+        input.setInputType(InputType.TYPE_CLASS_NUMBER);
+        InputFilter[] filterArray = new InputFilter[1];
+        filterArray[0] = new InputFilter.LengthFilter(4);
+        input.setFilters(filterArray);
+        mBuilder.setView(input);
+
+        mBuilder.setPositiveButton(getResources().getString(R.string.label_ok), new DialogInterface.OnClickListener() {
+                @Override
+                public void onClick(DialogInterface dialoginterface, int i) {
+                    OpenScale.getInstance().setBluetoothDeviceUserConsent(appUserId, consentCode[0], callbackBtHandler);
```

Comment:
Maybe we could set directly the consentCode preference here.

Reply:
See my response below: https://github.com/oliexdev/openScale/pull/753#discussion_r693802348